lib/src/Base/Stat/LatentVariableModel.cxx

```diff
@@ -0,0 +1,299 @@
+										 const UnsignedInteger latentDim)
+  : CovarianceModelImplementation(Point(1, 1.0), Point(1, 1.0))
+  , latentDim_(latentDim)
+  , nLevels_(nLevels)
+  , latentCovarianceMatrix_(nLevels_)
+  , latentCovarianceModel_(latentDim_)
+{
+  if (latentDim_ < 1) throw InvalidArgumentException(HERE) << "Error: the dimension of the latent space must be >= 1";
+  if (nLevels_ < 2) throw InvalidArgumentException(HERE) << "Error: the number of discrete levels must be >= 2";
+  activeLatentCoordinateDim_ = 1 + latentDim_ * (nLevels_ - 2);
+  activeLatentVariables_ = Point(activeLatentCoordinateDim_, 0.0);
+  fullLatentVariables_ = Sample(nLevels_, latentDim_);
+  Indices activeParameter = Indices(inputDimension_ + outputDimension_ + activeLatentCoordinateDim_);
+  activeParameter.fill();
+  setActiveParameter(activeParameter);
+  updateLatentCovarianceMatrix();
+}
+
+/* Virtual constructor */
+LatentVariableModel * LatentVariableModel::clone() const
+{
+  return new LatentVariableModel(*this);
+}
+
+/* Computation of the covariance  function */
+Scalar LatentVariableModel::computeAsScalar(const Scalar z1, const Scalar z2) const
+{
+  /*checks z1 and z2 being among the known levels*/
+  const bool isLevelz1 = (z1 >= 0 && z1 < nLevels_ && (z1 - floor(z1)) == 0);
+  const bool isLevelz2 = (z2 >= 0 && z2 < nLevels_ && (z2 - floor(z2)) == 0);
+
+  if (!isLevelz1 || !isLevelz2) throw InvalidArgumentException(HERE) << "Error: the input discrete variables values: " << z1 
+  << ", and/or " << z2 << " are not among the known levels. They should both present integer values between 0 and l-1.";
+
+  Scalar cov = latentCovarianceMatrix_(z1, z2);
+
+  return cov;
+}
+
+Scalar LatentVariableModel::computeAsScalar(const Point & z1, const Point & z2) const
+{
+  return computeAsScalar(z1[0], z2[0]);
+}
+
+
+Scalar LatentVariableModel::computeAsScalar(const Collection<Scalar>::const_iterator & z1_begin,
+    const Collection<Scalar>::const_iterator & z2_begin) const
+{
+
+  Collection<Scalar>::const_iterator z1_it = z1_begin;
+  Collection<Scalar>::const_iterator z2_it = z2_begin;
+
+  return computeAsScalar(*z1_it, *z2_it);
+}
+
+
+void LatentVariableModel::setFullParameter(const Point & parameter)
+{
+  /*
+    Care! To make the method not bogus, the size of parameter argument
+    should be :
+     - Size of scale : here 1
+     - Size of amplitude : here 1
+     - Number of latent variable coordinates : latentDim_ * nLevels_ - 2 * latentDim_ + 1
+    CovarianceModelImplementation::setFullParameter checks that size is
+    equal to the total number of parameters : 2 + the number of latent variable coordinates
+  */
+  // Check the size
+  const UnsignedInteger totalSize = 2 + activeLatentCoordinateDim_;
+  if (!(parameter.getSize() == totalSize))
+    throw InvalidArgumentException(HERE) << "In LatentVariableModel::setFullParameter, points have incompatible size. Point size = " << parameter.getSize()
```

Comment:
The information within the message is very good.
I think, however, that the macro `HERE` already gives the location of the error within the call stack, isn'it? So I guess that the name of the method `LatentVariableModel::setFullParameter` is not necessary in the message itself.
```suggestion
    throw InvalidArgumentException(HERE) << "The parameter has size = " << parameter.getSize()
```
Perhaps you could write a Python script which generates the exception and show the printed message as a reply to this discussion so that we can see what is actually printed.

Reply:
With the current version of the code, we have:

```
import openturns.experimental as otexp
kLat = otexp.LatentVariableModel(2,2)
kLat.setFullParameter([1.]*10)
```

output:
_In LatentVariableModel::setFullParameter, points have incompatible size. Point size = 10 whereas expected size = 3_

Therefore I am guessing the macro HERE does not behave like you think